clang/test/Driver/sycl-int-footer.cpp

```diff
@@ -18,9 +18,15 @@
 
 /// Preprocessed file use with integration footer
 // RUN: touch %t.ii
-// RUN:  %clangxx -fsycl -fsycl-use-footer %t.ii -### 2>&1 \
+// RUN:  %clangxx -fsycl %t.ii -### 2>&1 \
 // RUN:   | FileCheck -check-prefix FOOTER_PREPROC_USE %s
 // FOOTER_PREPROC_USE: clang-offload-bundler{{.*}} "-outputs=[[HOST1:.+\.ii]],[[DEVICE_PP:.+\.ii]]"
 // FOOTER_PREPROC_USE: clang{{.*}} "-fsycl-is-device"{{.*}} "[[DEVICE_PP]]"
 // FOOTER_PREPROC_USE: clang-offload-bundler{{.*}} "-outputs=[[HOST_PP:.+\.ii]],[[DEVICE1:.+\.ii]]"
 // FOOTER_PREPROC_USE: clang{{.*}} "-fsycl-is-host"{{.*}} "[[HOST_PP]]"
+
+/// Check that integration footer can be disabled
+// RUN:  %clangxx -fsycl -fno-sycl-use-footer %s -### 2>&1 \
+// RUN:   | FileCheck -check-prefix NO-FOOTER --implicit-check-not "-fsycl-int-footer" %s
+// NO-FOOTER: clang{{.*}} "-fsycl-is-device"{{.*}} "-fsycl-int-header=[[INTHEADER:.+\.h]]" "-sycl-std={{.*}}"
+// NO-FOOTER: clang{{.*}} "-include" "[[INTHEADER]]"{{.*}} "-fsycl-is-host"{{.*}} "-o"
```

Comment:
Adding a `-ccc-print-phases` check for `-fno-sycl-use-footer` should at least keep things in line in regards to driver expectations.  We should have at least some tests at the E2E level that take advantage of disabling the use of the footer if that is going to be a common scenario among users.

Reply:
> Adding a `-ccc-print-phases` check for `-fno-sycl-use-footer` should at least keep things in line in regards to driver expectations. 

Ok, will add

> We should have at least some tests at the E2E level that take advantage of disabling the use of the footer if that is going to be a common scenario among users.

It might make sense, but it is not expected to be a main usage model - the idea of the new flag is to have a workaround in case something go wrong with integration footer in post-commit or somewhere else. Most likely the option will be removed in the long run as more features added that depend on integration footer

Reply:
Added compiler phases test in 8e58b19. Will either add E2E test into llvm-test-suite or won't add it at all due to temporary/debug nature of `-fno-sycl-use-footer` option